guides/micronaut-static-resources/java/src/main/java/example/micronaut/MessageService.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.micronaut;
+
+import io.micronaut.core.annotation.NonNull;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+
+interface MessageService {
+
+    String sayHello(@NonNull @NotNull @NotBlank String name);
```

Comment:
Remove `@NotNull`. `@NotBlank` already checks whether the value is null or an empty string. 

```suggestion
    String sayHello(@NonNull @NotBlank String name);
```

Reply:
Done in https://github.com/micronaut-projects/micronaut-guides/pull/1426/commits/798c927efcca0e9d3fd3f9f863549c96aab4c9a8